open-metadata-implementation/access-services/asset-catalog/asset-catalog-server/src/main/java/org/odpi/openmetadata/accessservices/assetcatalog/handlers/CommonHandler.java

```diff
@@ -79,15 +92,113 @@ public String getTypeDefGUID(String userId, String typeDefName) {
     /**
      * Fetch the zone membership property
      *
-     * @param properties asset properties
+     * @param classifications asset properties
      * @return the list that contains the zone membership
      */
-    public List<String> getAssetZoneMembership(InstanceProperties properties) {
+    public List<String> getAssetZoneMembership(List<Classification> classifications) {
         String methodName = "getAssetZoneMembership";
-        List<String> zoneMembership = repositoryHelper.getStringArrayProperty(ASSET_CATALOG_OMAS, "zoneMembership", properties, methodName);
-        if (CollectionUtils.isNotEmpty(zoneMembership)) {
-            return zoneMembership;
+        if (CollectionUtils.isEmpty(classifications)) {
+            return Collections.emptyList();
+        }
+
+        Optional<Classification> assetZoneMembership = classifications.stream()
+                .filter(classification -> classification.getName().equals(ASSET_ZONE_MEMBERSHIP))
+                .findFirst();
+
+        if (assetZoneMembership.isPresent()) {
+            List<String> zoneMembership = repositoryHelper.getStringArrayProperty(ASSET_CATALOG_OMAS,
+                    "zoneMembership", assetZoneMembership.get().getProperties(), methodName);
+
+            if (CollectionUtils.isNotEmpty(zoneMembership)) {
+                return zoneMembership;
+            }
         }
+
         return Collections.emptyList();
     }
+
+    /**
+     * Return the requested entity, converting any errors from the repository services into the local
+     * OMAS exceptions.
+     *
+     * @param userId calling user
+     * @param guid   unique identifier for the entity
+     * @return entity detail object
+     * @throws InvalidParameterException  one of the parameters is null or invalid.
+     * @throws UserNotAuthorizedException user not authorized to issue this request.
+     * @throws PropertyServerException    problem retrieving the entity.
+     */
+    public EntityDetail getEntityByGUID(String userId,
+                                        String guid,
+                                        String type) throws InvalidParameterException, UserNotAuthorizedException, PropertyServerException {
+
+        String entityTypeName = type == null ? "Unknown" : type;
```

Comment:
type is supplied but does not seem to be used. We should either use the type or remove it from the method signature.